### DISCUSSION TITLE
Cosori Dual Blaze 6.8-Quart Smart Air Fryer CAF-P583S

### DISCUSSION THREAD

**Discussion by webdjoe** [webdjoe/pyvesync#249] [Kitchen]
Discussion for Cosori Dual Blaze 6.8-Quart Smart Air Fryer CAF-P583S

#55

**Comment by epikarus** [comment]
Hi.  I see there has been quite of bit of effort to get this air fryer supported.  I am using home assistant and noticed that my this airfryer, which used to work on some of the custom vesync integrations no longer works. It's been a few months and I don't know when it stopped working.

If there is something I can do to get this ball back rolling on support for this device I would love to provide what I can.  My knowledge when it comes to these things is quite limited so there might be a lot of handholding, I apologize, but I'm willing to follow directions. Thank you.

**Comment by koshisan** [comment]
Same here - its absolutely maddening. It used to work a while ago with a custom integration (I cannot find which one). I switched for a while to Ninja Doublestack, but was never really satisfied with it. Now I switched back to the Cosori, but can't find a working integration anymore. I even patched some of the older forks myself to get them working with newer HA versions - but no luck...

**Comment by cdnninja** [comment]
Have you tried using just the library to see if it can control the device? It isn't listed so wouldn't expect it to work.   You could try it using this new version here just to confirm: https://github.com/webdjoe/pyvesync/tree/dev-2.0

**Comment by koshisan** [comment]
Unfortunately I cant get the dev-2.0 tree running and I also broke my fingers over the last two days sniffing out the stuff from the app itself - with no success unfortunately. HOWEVER: I did find a backup of my old HA instance where I had the integration running and the Manifest points to a fork "cosmin-cucu-pyvesyncv2==0.0.3"

It does not seem to be on github, but I did find this: https://packages.ecosyste.ms/registries/pypi.org/packages/cosmin-cucu-pyvesyncv2?path=

And when you look into the vesynckitchen.py, it indeed does have a declaration:

    'CosoriCAF': {
        'module': 'VeSyncAirFryerCAF',
        'models': ['CAF-P583S-KUS', 'CAF-P583S-KEU', 'CAF-P681S-KEU', 'CAF-P681S', 'CAF-P681S-KEUR', 'CAF-LI401S'],
        'features': [],
    },

**Comment by webdjoe** [comment]
I will add this to the list for V3. We are close to pushing out the first official release of V3, I will work on this when we release.

**Comment by samuel-zamora** [comment]
Hola, me sumo a la petición ya que veo que hay comentarios recientes jajaj yo también tengo una dual Blaze y estoy empezando en HA y veo que no aparece con la integración oficial de Vesync, así que me ofrezco también para ayudar en lo que pueda.

Un saldudo desde España

**Comment by anoblet** [comment]
Any update on this? I appreciate all of the work!

**Comment by mikealanni** [comment]
I have the same fryer but I cannot add it to home assistant as the integration worked but no devices as its says it's now available in device map in the logs

**Comment by webdjoe** [comment]
This needs to be added to the pyvesync library before it can be integrated with HA. In order to do that, I'll need comprehensive packet captures. Follow the directions https://webdjoe.github.io/pyvesync/latest/development/capturing/ or you can share the device with me temporarily and I will capture the packets. 

Apologies for the run around, but let's keep this discussion in https://github.com/webdjoe/pyvesync/issues/477
